File: testdata/go/rsa_encrypt.go
package main

import (
	"crypto/rand"
	"crypto/rsa"
	"crypto/sha256"
	"fmt"
)

func main() {
	// Generate RSA key pair - QUANTUM VULNERABLE
	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
	if err != nil {
		panic(err)
	}

	message := []byte("secret data")

	// Encrypt with RSA-OAEP - vulnerable to Shor's algorithm
	ciphertext, err := rsa.EncryptOAEP(sha256.New(), rand.Reader, &privateKey.PublicKey, message, nil)
	if err != nil {
		panic(err)
	}

	// Decrypt
	plaintext, err := rsa.DecryptOAEP(sha256.New(), rand.Reader, privateKey, ciphertext, nil)
	if err != nil {
		panic(err)
	}

	fmt.Println(string(plaintext))
}

File: testdata/python/rsa_encrypt.py
from cryptography.hazmat.primitives.asymmetric import rsa
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.asymmetric import padding

# Generate RSA key - QUANTUM VULNERABLE
private_key = rsa.generate_private_key(
    public_exponent=65537,
    key_size=2048,
)

message = b"secret data"

# Encrypt with RSA - vulnerable to Shor's algorithm
ciphertext = private_key.public_key().encrypt(
    message,
    padding.OAEP(
        mgf=padding.MGF1(algorithm=hashes.SHA256()),
        algorithm=hashes.SHA256(),
        label=None
    )
)
